plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codetest/CodeTestException.kt

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.codetest
+
+import software.aws.toolkits.resources.message
+
+open class CodeTestException(
+    override val message: String?,
+    val statusCode: String? = "400",
+    val code: String? = "DefaultError",
+    val uiMessage: String? = message(
+        "testgen.error.generic_error_message"
+    ),
+) : RuntimeException()
+
+internal fun noFileOpenError(): Nothing =
+    throw CodeTestException(message("codewhisperer.codescan.no_file_open"), "400", "ProjectZipError")
+
+internal fun fileTooLarge(): Nothing =
+    throw CodeTestException(message("codewhisperer.codescan.file_too_large_telemetry"), "400", "ProjectZipError")
+
+internal fun cannotFindFile(errorMessage: String, filepath: String): Nothing =
+    error(message("codewhisperer.codescan.file_not_found", filepath, errorMessage))
+
+internal fun cannotFindValidFile(errorMessage: String): Nothing =
+    throw CodeTestException(errorMessage, "400", "ProjectZipError")
+
+internal fun cannotFindBuildArtifacts(errorMessage: String): Nothing =
+    throw CodeTestException(errorMessage, "400", "ProjectZipError")
+
+internal fun invalidSourceZipError(): Nothing =
+    throw CodeTestException(message("codewhisperer.codescan.invalid_source_zip_telemetry"), "400", "InvalidSourceZipError")
+
+fun codeTestServerException(
+    errorMessage: String,
+    statusCode: String?,
```

Comment:
status code should always be a number

Reply:
+1

Reply:
statusCode is string in MetricBase so to align with VSC and JB I initialized this to String.

Reply:
Thought of using number but in our Telemetry statusCode is declared as String.
[aws-toolkit-common](https://github.com/aws/aws-toolkit-common/blob/5ebdd76f5dd424c35def8eef174fd837d5978819/telemetry/csharp/AwsToolkit.Telemetry.Events/Core/BaseTelemetryEvent.cs#L46-L50)

If required I can use number here and can convert to string while emitting?

Reply:
yes please

Reply:
Addressed.